doc/neptune-metadata.rst

```diff
@@ -0,0 +1,105 @@
+===========================
+Neptune Metadata Definition
+===========================
+
+This file specifies what fields are in the ``.neptune.yml`` file, what they mean
+and by which tools they are used. The file itself is expected to be in a
+repositories root directory.
+
+.. contents::
+
+library
+-------
+
+Type
+  boolean
+Meaning
+  If set to true, the repository is marked as a library used by other repositories
+Tools
+  - neptune-overview
+  - neptune-autopr
+Example
+  .. code:: yml
+
+    library: true
+
+d2ready
+-------
+
+Type
+  boolean
+Meaning
+  If set to true, the repository is marked as ready for D2 compilation
+Tools:
+  - neptune-overview
+Example
+  .. code:: yml
+
+    d2ready: true
+
+support-guarantees
+------------------
+
+Type
+  Mapping
+Meaning
+  Sets the support window and period users can expect for this repository
+Tools:
+  - neptune-overview
+  - neptune-autopr
+Example
+  .. code:: yml
+
+    support-guarantees:
+      major-months: 6
+      minor-versions: 2
+
+major-months
+~~~~~~~~~~~~
+
+Type
+  Integer
+Meaning
+  Amount of months a major version is supported, starting from the time the next
+  major version is released
+
+minor-versions
+~~~~~~~~~~~~~~
+
+Type
+  Integer
+Meaning
+  Amount of minor versions that are supported, starting from the latest one
+  
+auto-patch-level-global
```

Comment:
It may already have been decided (in the tool), but I wonder about including the word "patch" in this name. (Ditto for `auto-patch-level`, below.) One of the _possible_ values is `Patch`, so I suspect the variable itself isn't purely related to patches. `auto-update`? `auto-pr`?

Reply:
`request-automatic-upgrade-prs`

Reply:
Nothing is decided yet